### PR TITLE
Jcs/reports20190402

### PR DESCRIPTION
**Pull request by jSylvestre**
This PR includes the api for the same info

### CODE REVIEW

**Comment by jSylvestre** [comment]
image

image

**Comment by jSylvestre** [comment]
Part of issue #546

**Comment by jcubbage** [comment]
The only thing I'd say is our list of API links at TEAM/TeamAdmin/RegenerateApiCode is going to get really big. We should pull that out to a new page to list out all the feeds teams will eventually have.

File: Keas.Mvc/Controllers/FeedController.cs [hunk 1]
```diff
@@ -8,18 +8,21 @@
...
+         private readonly IReportService _reportService;
 
-        public FeedController(ApplicationDbContext context)
-        {
-            this._context = context;
-        }
+         public FeedController(ApplicationDbContext context, IReportService reportService)
+         {
+             this._context = context;
+             _reportService = reportService;
```
Comment (srkirkland) [inline Keas.Mvc/Controllers/FeedController.cs]: why use `this._context` one line up but not `this._reportService` on this line?

File: Keas.Mvc/Controllers/ReportController.cs [hunk 2]
```diff
@@ -19,11 +19,13 @@ public class ReportController : SuperController
...
 
         private readonly ISecurityService _securityService;
+        private readonly IReportService _reportService;
 
-        public ReportController(ApplicationDbContext context, ISecurityService securityService)
+        public ReportController(ApplicationDbContext context, ISecurityService securityService, IReportService reportService)
         {
             this._context = context;
             this._securityService = securityService;
+            _reportService = reportService;
```
Comment (srkirkland) [inline Keas.Mvc/Controllers/ReportController.cs]: same here -- you should change to make them match

File: Keas.Mvc/Services/ReportService.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+
+        public async Task<IList<WorkstationReportModel>> WorkStations(Team team, string teamSlug)
+        {
+            if (team == null)
+            {
+                team = await _context.Teams.SingleAsync(a => a.Slug == teamSlug);
+            }
+
+            return await _context.Workstations.Include(a => a.Space).Include(a => a.Assignment).ThenInclude(a => a.Person).IgnoreQueryFilters().AsNoTracking().Where(a => a.TeamId == team.Id).Select(a => new WorkstationReportModel
```
Comment (srkirkland) [inline Keas.Mvc/Services/ReportService.cs]: try getting rid of the `.Include(...)` parts of this and run again.  I don't think you'll need them and I think they might be grabbing more info than you need.  You don't need the `.Include()` if you are projecting out with a `.Select()`

File: Keas.Mvc/Views/Report/Index.cshtml [hunk 4]
```diff
@@ -63,5 +63,18 @@
     <div class="card-content">
         <a asp-action="PersonActions" class="btn btn-link">Actions taken against people in team <i class="fas fa-chevron-right fa-sm" aria-hidden="true"></i></a>
     </div>
-</div>
+</div>
+ @if ((await AuthorizationService.AuthorizeAsync(User, AccessCodes.Codes.SpaceMasterAccess)).Succeeded)
```
Comment (srkirkland) [inline Keas.Mvc/Views/Report/Index.cshtml]: this is ok but we should come up with a better way of authorizing reports, or cache this.  Currently the report page is going to be doing 2 queries per report so we are probably up to 10 already, just to figure out what the user can access.  One idea is to query for the user's roles server side and then use that to check if user isInRole inside the page.  Something to keep in mind anyway

File: Keas.Mvc/Controllers/FeedController.cs (hunk 1 above)
Comment (jSylvestre) [reply Keas.Mvc/Controllers/FeedController.cs]: Don't need the this. I'll remove that in both places 
Blame @jcubbage  :)

**Pull request merged**